doc/sfp-cmis/Interface-Link-bring-up-sequence.md

```diff
@@ -0,0 +1,195 @@
+  - CMIS complaint optical modules:-
+      All CMIS complaint optical modules will follow this approach as recommended in the CMIS spec.
+  - SFF complaint optical modules:- 
+    - deterministic approach to bring the interface will eliminate any link stability issue which will be difficult to chase in the production network
+      e.g. If there is a PHY device in between, and this 'deterministic approach' is not followed, PHY may adapt to a bad signal or interface flaps may occur when the optics tx/rx  enabled during PHY initialization. 
+    - there is a possibility of interface link flaps with non-quiescent optical modules <QSFP+/SFP28/SFP+> if this 'deterministic approach' is not followed
+    - It helps bring down the optical module laser when interface is adminstiratively shutdown. Per the workflow here, this is acheived by xcvrd listening to host_tx_ready field from PORT_TABLE of STATE_DB. Turning the laser off would reduce the power consumption and avoid any lab hazard
+    - Additionally provides uniform workflow (from SONiC NOS) across all interface types with or without module presence. 
+  - This synchronization will also benefit SFP+ optical modules as they are "plug N play" and may not have quiescent functionality. (xcvrd can use the optional 'soft tx disable' ctrl reg to disable the tx)
+
+# Pre-requisite
+
+As mentioned above in 'Background' and 'Plan' sections, need to follow specified bring-up sequence.
+Work flows are designed considering SONiC NOS operating in sync mode.
+
+In case SONiC NOS operates in async mode, then expected behavior is - the return status of the set ADMIN_STATE attribute update in ASIC-DB (syncd/GBsyncd) will be treated to set the host_tx_ready in Orchagent.
+
+# Breakout Handling
+  - The new 'host_tx_ready' field of Port table in state-DB is created for every interface <regular/breakout interface>.
+  - Xcvrd processes the 'host_tx_ready' change event and is responsible to disable Tx/laser for all optical lanes or respective optical lane that belongs to the interface in case of breakout.
+  - Currently the logical mapping between the interface and optical lane is not present in xcvrd. Creating this logical mapping in xcvrd will address breakout interface handling.
+
+# Proposed Work-Flows
+
+Please refer to the  flow/sequence diagrams which covers the following required use-cases
+  - Enabling this feature 
+  - Transceiver initialization
+  - admin enable configurations 
+  - admin disable configurations
+  - No transceiver present
+
+# Feature enablement
+  This feature (optics Interface Link bring-up sequence) would be enabled on per platform basis.
+  There could be cases where vendor(s)/platform(s) may take time to shift from existing codebase to the model (work-flows) described in this document.
+  In order to avoid any breakage and ensure gradual migration of different platforms/vendors to this model, there would be new field (flag) in xcvrd to enable/disable this feature. 
```

Comment:
Would you please specify "is_link_bringup_seq_deterministic" mentioned in the workflow chart? Is it a new platform API? but from the description in the chart seems it's defined in C instead of Python.

Reply:
>>  Is it a new platform API? 
Yes - its a new platform (plugin) API to supply/update xcvrd whether platform (hwsku) wants to make use of this feature or not.
>> but from the description in the chart seems it's defined in C instead of Python.
It would be python

Reply:
what do you mean by "plugin"?  if you are referring to the plugin which resides in the vendor platform folder like https://github.com/Azure/sonic-buildimage/tree/master/device/mellanox/x86_64-mlnx_msn2700-r0/plugins , I think they are deprecated by the platform API and should not be used.

Reply:
Updated the workflow in HLD. This would be determined via ‘skip_xcvrd_cmis_mgr’ of pmon_daemon_control.json
and in turn parsed by xcvrd to determine whether to launch CMIS task manager (i.e. enable this feature) or not (i.e. leave/keep this feature didabled)